### DISCUSSION TITLE
Audacity doesn't pick up the signal from blackhole on MacOS Ventura

### DISCUSSION THREAD

**Discussion by ip60** [audacity/audacity#4898] [Q&A] [answered]
### Bug description

I've installed the last version of Audacity (3.3.3 for Apple Silicon) on the last version of macOS 13.4 and installed the last version of Blackhole (0.5.0) to record the sound from the computer. Blackhole does perfectly his job recording the sound from the computer when using any other app but somehow, I don't get audacity to pick up the signal from blackhole. I've tried transport > rescan audio devices, I've tried to reboot, it still doesn't work. And it's only a problem in Audacity. At other times I did get an error though, when instead of using the multi-output virtual device, I just try to output to the speakers. 

Here attached are my audio settings and the error I get (full text follows below)
<img width="609" alt="audacity audio settings" src="https://github.com/audacity/audacity/assets/95503317/2ef408ce-90f1-4b43-9161-30b1ba8fba44">
<img width="396" alt="Audacity error -9986" src="https://github.com/audacity/audacity/assets/95503317/050d3717-9d2a-4cb4-aea4-2bb920379ac5">

Full text in the error is : {
    "timestamp": 1687611144,
    "event_id": "fe3da171093a4425b01ddeddca2f4fbf",
    "platform": "native",
    "release": "audacity@3.3.3",
    "contexts": {
        "os": {
            "type": "os",
            "name": "Macintosh",
            "version": "10.16.0"
        }
    },
    "exception": {
        "values": [
            {
                "type": "Error",
                "value": "Error opening recording device.\nError code: -9986 Internal PortAudio error.",
                "mechanism": {
                    "type": "runtime_error",
                    "handled": false
                }
            }
        ]
    }
}

### Steps to reproduce

So I just try to record the computer sounds (actually a video playing) with Blackhole and it either doesn't pick anything or it throws out an error. 

### Expected behavior

All the other apps (quicktime, screen recorder) can pick up and record the sound, somehow that doesn't work with audacity on Apple silicon with macOS ventura 13.4. All apps were the latest versions. 

### Actual behavior

See above. I'm repeating myself

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

macOS

### Additional context

Only valuable context is that Blackhole works well with other recording apps, it's an audacity issue.

**Comment by LWinterberg** [comment]
does it work if you record 16 channels instead of just stereo?

**Comment by ip60** [comment]
No I get 16 flat lines instead of just two. No sound picked up from the computer.

**Comment by ip60** [comment, marked as answer]
I found how to make it work: in the settings panel > privacy & Security you have to click on microphone and allow audacity to access the microphone. And then it suddenly started working. But this is seriously counter-intuitive as I had already considered this and thought that has nothing to do with the virtual device and that wouldn't work. This discussion can be closed as far as I am concerned.
